geometry/scalar.h

```diff
@@ -5,13 +5,24 @@
 #pragma once
 
 #include <cfloat>
+#include <limits>
+#include <valarray>
 
+#include "flutter/fml/logging.h"
 #include "impeller/geometry/constants.h"
 
 namespace impeller {
 
 using Scalar = float;
 
+constexpr bool ScalarNearlyEqual(
+    Scalar x,
+    Scalar y,
+    Scalar tolerance = std::numeric_limits<Scalar>::epsilon() * (1 << 16)) {
+  FML_DCHECK(tolerance >= 0);
```

Comment:
Perhaps just clamp it 0 instead of a DCHECK which will be ignored in most builds?

Reply:
Clamping to zero would have the same effect as just removing the check. If someone makes a negative tolerance it seems like that'd be unintentional, but it's not like it's a real crasher. Maybe I should just remove it?